lib/kill.js

```diff
@@ -1,34 +1,36 @@
 import os from 'node:os';
-import {setTimeout as pSetTimeout} from 'node:timers/promises';
+import {setTimeout} from 'node:timers/promises';
 import {onExit} from 'signal-exit';
 
 const DEFAULT_FORCE_KILL_TIMEOUT = 1000 * 5;
 
 // Monkey-patches `childProcess.kill()` to add `forceKillAfterTimeout` behavior
-export const spawnedKill = (kill, signal = 'SIGTERM', {forceKillAfterTimeout = true} = {}) => {
+export const spawnedKill = (kill, {forceKillAfterTimeout}, controller, signal) => {
 	const killResult = kill(signal);
-	const timeout = getForceKillAfterTimeout(signal, forceKillAfterTimeout, killResult);
-	setKillTimeout(kill, timeout);
+	setKillTimeout({kill, signal, forceKillAfterTimeout, killResult, controller});
 	return killResult;
 };
 
-const setKillTimeout = async (kill, timeout) => {
-	if (timeout === undefined) {
+const setKillTimeout = async ({kill, signal, forceKillAfterTimeout, killResult, controller}) => {
+	if (!shouldForceKill(signal, forceKillAfterTimeout, killResult)) {
 		return;
 	}
 
-	await pSetTimeout(timeout, undefined, {ref: false});
-	kill('SIGKILL');
+	try {
+		await setTimeout(forceKillAfterTimeout, undefined, {signal: controller.signal});
```

Comment:
The controller is aborted when the child process exits (successfully or not). When this happens, `setTimeout()` throws, hence the empty `catch` block.